genesyscloud/outbound_callabletimeset/resource_genesyscloud_outbound_callabletimeset.go

```diff
@@ -0,0 +1,139 @@
+func getAllOutboundCallableTimesets(ctx context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	resources := make(resourceExporter.ResourceIDMetaMap)
+	proxy := getOutboundCallabletimesetProxy(clientConfig)
+
+	callabletimesets, getErr := proxy.getAllOutboundCallableTimeset(ctx)
+	if getErr != nil {
+		return nil, diag.Errorf("Failed to get page of callable timeset configs: %v", getErr)
+	}
+	for _, callabletimesets := range *callabletimesets {
+		resources[*callabletimesets.Id] = &resourceExporter.ResourceMeta{Name: *callabletimesets.Name}
+	}
+	return resources, nil
+}
+
+// createOutboundCallabletimeset is used by the outbound_callabletimeset resource to create Outbound Callable Timesets
+func createOutboundCallabletimeset(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := newOutboundCallableTimesetProxy(sdkConfig)
+
+	callableTimeset := platformclientv2.Callabletimeset{
+		CallableTimes: buildCallableTimes(d.Get("callable_times").(*schema.Set)),
+	}
+
+	if name != "" {
+		callableTimeset.Name = &name
+	}
+
+	log.Printf("Creating Outbound Callabletimeset %s", name)
+	outboundCallabletimeset, err := proxy.createOutboundCallabletimeset(ctx, &callableTimeset)
+	if err != nil {
+		return diag.Errorf("Failed to create Outbound Callabletimeset %s: %s", name, err)
+	}
+
+	d.SetId(*outboundCallabletimeset.Id)
+
+	log.Printf("Created Outbound Callabletimeset %s %s", name, *outboundCallabletimeset.Id)
+	return readOutboundCallabletimeset(ctx, d, meta)
+}
+
+// updateOutboundCallabletimeset is used by the outbound_callabletimeset resource to update an Outbound Callable Timeset
+func updateOutboundCallabletimeset(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := newOutboundCallableTimesetProxy(sdkConfig)
+
+	callableTimeset := getOutboundCallableTimesetFromResourceData(d)
+
+	log.Printf("Updating Outbound Callabletimeset %s", d.Id())
+	outboundCallabletimeset, err := proxy.updateOutboundCallabletimeset(ctx, d.Id(), &callableTimeset)
+	if err != nil {
+		return diag.Errorf("Failed to read Outbound Callabletimeset %s: %s", d.Id(), err)
+	}
+
+	log.Printf("Updated Outbound Callabletimeset %s", outboundCallabletimeset.Id)
+	return readOutboundCallabletimeset(ctx, d, meta)
+}
+
+// readOutboundCallabletimeset is used by the outbound_callabletimeset resource to read an Outbound Callable Timeset from the genesys cloud
+func readOutboundCallabletimeset(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := newOutboundCallableTimesetProxy(sdkConfig)
```

Comment:
Use getOutboundCallabletimesetProxy instead of the newOutboundCallableTimesetProxy() function